docs/development/workspace-file-system-architecture.md

```diff
@@ -0,0 +1,155 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+# Workspace file system architecture
+
+*this is a work in progress - what's presented here may not correspond to the current state of the code base*
+
+Several modules can affect the file system of a given workspace:
+
+- the fuse mountpoint
+- the WinFSP mountpoint
+- the path interface
+- the synchronization module
+
+Each of these modules relies on the transactions provided by the workspace file system. Those transactions have the following properties:
+
+- **Atomicity**: they either fail or succeed, but do not leave the system in an inconsistent or transitional state.
+- **Safety**: they are protected internally, so several transactions can be executed concurrently.
+- **Locality**: they only affect a single entry of the file system (more specifically, a single manifest).
+- **Agnostic**: they behave the same regardless of the current platform.
+
+## List of file system transactions
+
+The following transactions target entries represented by their file system path:
+
+### Entry transactions
+
+- `entry_info`: returns a dictionary with all the information about the given entry.
+- `entry_rename` (overwrite=True): rename the given entry, possibly overwriting its destination.
+- `entry_rename` (overwrite=False): rename the given entry, failing if the destination already exists.
+
+> It is not possible to rename to another directory, as this would break the locality principle.
+> These operations affect the parent of the entry to rename and not the entry itself.
```

Comment:
```suggestion
> It is not possible to rename to another directory, as this would break the locality principle.

> These operations affect the parent of the entry to rename and not the entry itself.
```

Reply:
I've reformatted the document. This suggestion is deprecated